core/trino-main/src/main/java/io/trino/operator/OutputSpoolingOperatorFactory.java

```diff
@@ -254,18 +256,18 @@ public boolean isFinished()
             return state == FINISHED;
         }
 
-        private Page outputBuffer(boolean finished)
+        private Page outputBuffer()
         {
             if (buffer.isEmpty()) {
                 return null;
             }
 
             synchronized (buffer) {
-                return spool(buffer.removeAll(), finished);
+                return spool(buffer.removeAll());
```

Comment:
can you add `@GuardedBy` annotations so it is obviouls what is the synchronization for and where it is needed?